api/src/database/run-ast.ts

```diff
@@ -476,21 +476,18 @@ function mergeWithParentItems(
 
 			parentItem[nestedNode.fieldKey].push(...itemChildren);
 
+			const limit = nestedNode.query.limit ?? Number(env['QUERY_LIMIT_DEFAULT']);
```

Comment:
Do we need to account for `QUERY_LIMIT_MAX` here or is that handled by the `nestedNode.query.limit` upstream? Dont see it in the old implementation either so probably the latter 🤔

Reply:
> Dont see it in the old implementation either so probably the latter

Yeah, should be checked/applied at higher level 👍 This change only ensures the check before slicing also accounts for the the fallback (default) value.